.github/workflows/main.yml

```diff
@@ -37,87 +37,57 @@ jobs:
       - run: echo "::add-matcher::.github/workflows/python-warning-matcher.json"
         name: "Setup GitHub for reporting Python warnings as annotations in pull request code review"
 
-      - uses: pre-commit/action@v3.0.0
-        name: Run pre-commit checks (no spaces at end of lines, etc)
-        if: ${{ matrix.python-version != '2.7' }}
+      # https://github.com/marketplace/actions/run-pyright-with-reviewdog (annotations only)
+      - uses: jordemort/action-pyright@v1
```

Comment:
How does this action interact with the pre-commit hook you've configured? I thought the workflow would cal the pre-commit to ensure what's run locally is the same as what's run in the workflow

Reply:
I'm also not fond of the author's lack of explaining why is the github secret needed. I guess it's to add comments, but this needs to be explained in the original repository

Reply:
Like with `pylint`, it would have been check, that would only show something if local CI was not done before.

The GitHub token is short-lived and while the attitude of the developer of just saying that it is needed is not very helpful, he stated just what he knew to apply. It does not mean that he was offensive, even if it may look like.